Add cli arg include-simulators

kandji-joel opened the pull request:
Closes #443 

`appium server -ka 800 --use-plugins=device-farm,appium-dashboard -pa /wd/hub --plugin-device-farm-platform=ios --plugin-device-farm-include-simulators=false`

<img width="1147" alt="image" src="https://user-images.githubusercontent.com/106340952/178114202-39a9c64a-0398-4e86-b7f5-f8af423f250a.png">

saikrishna321 commented:
@kandji-joel Thanks for the PR. Can you please add Unit tests for this?

saikrishna321 commented:
@kandji-joel Can you please fix the CI. Also, can you add tests which filter only simulators.

kandji-joel commented:
@saikrishna321 I fixed CI issues and added some tests -- I haven't wrote those style tests before so let me know you'd like to see some changes

>Also, can you add tests which filter only simulators.

With the above implementation, you can only exclude simulators from real devices. You cannot exclude real devices from simulators. If you think it would be equally beneficial to exclude real devices I could change the argument from `--plugin-device-farm-include-simulators=true(default)|false` to something like `--plugin-device-farm-device-types=both(default)|simulated|real`. Thoughts?

SrinivasanTarget commented:
@sudharsan-selvaraj agree with you. Probably we can merge this as-is and improve the same over the next set of PRs. @kandji-joel what do you think?

saikrishna321 commented:
Happy to merge! As @SrinivasanTarget mentioned we can handle it in another PR and also move these CLI args to server config file. 

Thoughts @sudharsan-selvaraj?

sudharsan-selvaraj commented:
@saikrishna321 Sounds good. We can merge this PR then.

saikrishna321 commented:
@kandji-joel can you please fix conflict

kandji-joel commented:
@saikrishna321 @sudharsan-selvaraj sounds good to me, added an issue here https://github.com/AppiumTestDistribution/appium-device-farm/issues/455
